src/lib/bcmp/bcmp.cpp

```diff
@@ -208,31 +199,30 @@ BmErr bcmp_tx(const void *dst, BcmpMessageType type, uint8_t *data, uint16_t siz
     \return ERR_OK on success, or an error that occurred.
 */
 BmErr bcmp_ll_forward(BcmpHeader *header, void *payload, uint32_t size, uint8_t ingress_port) {
-  ip_addr_t port_specific_dst;
-  ip6_addr_set(&port_specific_dst, &multicast_ll_addr);
+  uint8_t port_specific_dst[sizeof(multicast_ll_addr)];
   BmErr err = BmEINVAL;
+  void *forward = NULL;
+  memcpy(port_specific_dst, &multicast_ll_addr, sizeof(multicast_ll_addr));
 
   // TODO: Make more generic. This is specifically for a 2-port device.
   uint8_t egress_port = ingress_port == 1 ? 2 : 1;
-  port_specific_dst.addr[3] = 0x1000000 | (egress_port << 8);
-
-  const ip_addr_t *src = netif_ip_addr6(CTX.netif, 0);
+  ((uint32_t *)port_specific_dst)[3] = 0x1000000 | (egress_port << 8);
 
   // L2 will clear the egress port from the destination address,
   // so calculate the checksum on the link-local multicast address.
-  struct pbuf *pbuf = pbuf_alloc(PBUF_IP, size + sizeof(bcmp_header_t), PBUF_RAM);
+  forward = bm_ip_tx_new(&multicast_ll_addr, size + sizeof(BcmpHeader));
   header->checksum = 0;
-  header->checksum = ip6_chksum_pseudo(pbuf, IP_PROTO_BCMP, pbuf->len, src, &multicast_ll_addr);
-  memcpy(pbuf->payload, header, sizeof(BcmpHeader));
-  memcpy((uint8_t *)pbuf->payload + sizeof(BcmpHeader), payload, size);
+  header->checksum = packet_checksum(forward, size + sizeof(BcmpHeader));
+
+  // Copy data to be forwarded
+  bm_ip_tx_copy(forward, header, sizeof(BcmpHeader), 0);
+  bm_ip_tx_copy(forward, payload, size, sizeof(BcmpHeader));
 
-  err = raw_sendto_if_src(CTX.pcb, pbuf, &port_specific_dst, CTX.netif, src) == ERR_OK
-            ? BmOK
-            : BmEBADMSG;
+  bm_ip_tx_perform(forward, &port_specific_dst);
   if (err != BmOK) {
     printf("Error forwarding BMCP packet link-locally: %d\n", err);
   }
-  pbuf_free(pbuf);
+  bm_ip_tx_cleanup(forward);
   return err;
```

Comment:
Looks like `err` is not being set anywhere in the new changes

Reply:
Will adjust